### ISSUE TITLE
Having a kornia.nn module to include all module-like APIs.

### ISSUE THREAD

**Issue by shijianjian** [kornia/kornia#520]
Currently, most APIs are scattered in different files. Also, we are mixing the functional APIs and module-like APIs. Should we have a ```kornia.nn``` to split the module-like APIs out of functional ones?

- For a simple starter, we may have only ```__init__.py``` inside the ```kornia.nn``` to create a shortcut for users. To have something like:
    ```python
    from kornia.augmentation import (
        RandomHorizontalFlip,
        RandomVerticalFlip,
        RandomErasing,
        RandomGrayscale,
        ColorJitter,
        RandomRotation,
        RandomCrop,
        RandomResizedCrop,
    )
    from kornia.color.normalize import (
        Normalize,
        Denormalize
    )
    from kornia.gray import (
        RgbToGrayscale,
        BgrToGrayscale
    )
    ...
    ```
- Eventually, to have a file structure like below to separate those two different APIs:

    - kornia
        - nn
            - agumentation
            - color
            - feature
            - filters
            - geometry
        - augmentation
        - color
        - feature
        - filters
        - geometry

**Comment by edgarriba** [comment]
seems a good idea/approach. /cc @ducha-aiki @lferraz

**Comment by yutayamazaki** [comment]
@edgarriba Can I work on this issue?

**Comment by edgarriba** [comment]
@yutayamazaki yes, please go ahead. Just we need to take care about potential cycle dependencies. /cc @shijianjian

**Comment by shijianjian** [comment]
@yutayamazaki are you still working on this? How's it going?

**Comment by edgarriba** [comment]
@shijianjian what's the approach we take on this ?

**Comment by cceyda** [comment]
Sounds related: https://github.com/kornia/kornia/issues/804

**Comment by JoanFM** [comment]
I think just having the module do all the `imports` would be enough.

 Maybe u can have some magic to do all the imports by doing inspection of the classes, so that it is easier to mantain.

**Comment by shijianjian** [comment]
I am thinking to resume this given the needs of using Kornia preprocessing/augmentation/operation as layers.

More or less, at least to have a summary list like:

| Operation | Diffrentiable | JIT | ONNX |
|---|------|------|------|
| GrayToRGB | Y | Y | Y |
| Equalize | N | Y | Y |
| Affine | Y | Y | N |
